engine/runtime/src/main/scala/org/enso/compiler/context/ChangesetBuilder.scala

```diff
@@ -42,8 +51,56 @@ final class ChangesetBuilder[A: TextEditor: IndexedSource](
     * @return the computed changeset
     */
   @throws[CompilerError]
-  def build(edits: Seq[TextEdit]): Changeset[A] =
-    Changeset(source, ir, compute(edits))
+  def build(edits: Seq[TextEdit]): Changeset[A] = {
+
+    def findSimpleUpdate(): SimpleUpdate = {
+      val edit: TextEdit = edits match {
+        case Seq(first) => first
+        case Seq(head, realEdit) => {
+          val firstAffected = invalidated(Seq(head))
+          if (!firstAffected.isEmpty) {
+            return null
```

Comment:
Can you rewrite it so as to avoid explicit returns?

Reply:
I assume this should also go away when we integrate the new API request for a slider change. So I'm ok with leaving it as is.

Reply:
Rewritten to `Option.map.filter.map.filter` in the latest changeset.